src/angular/app.js

```diff
@@ -98,6 +99,12 @@ services.factory('Search', ['ServicesList', function (ServicesList) {
         },
         currResults: function () {
             return metaDimension.top(Infinity);
+        },
+        setServicesById: function (servicesById) {
+        	this.servicesById = servicesById;
+        },
+        getServicesById: function () {
+        	return this.servicesById;
```

Comment:
I don't think this should be on the `Search` service since it's not related to search. Create a new service called `ServiceRepository` or something like that and add a method to it `findById`. This method will return the service.
This new service will take in the `ServicesList` as a dependency and use it to initialize the map from `serviceId: service`.